testng-core/src/main/java/org/testng/internal/objects/pojo/CreationAttributes.java

```diff
@@ -13,7 +12,6 @@ public class CreationAttributes {
   private final GuiceContext suiteContext;
 
   public CreationAttributes(ITestContext ctx, BasicAttributes basic, DetailedAttributes detailed) {
-    Objects.requireNonNull(ctx);
```

Comment:
Why did you remove this requirement?

Reply:
I removed this because this edit check prevented me from leveraging the `DefaultListenerFactory` for creating listener instances, when it comes to instantiating listeners for class names provided via the CLI and when a user does not provide a custom factory implementation. I basically wanted to try and push all the instantiation logic via the same code path.